packages/web-react/src/components/Field/ValidationText.tsx

```diff
@@ -28,15 +33,25 @@ export const ValidationText = <T extends ElementType = 'div'>(props: ValidationT
     };
   }, [id, registerAria]);
 
-  if (validationText) {
+  useEffect(() => {
```

Comment:
**suggestion (if-minor):** I think we should extract this logic into a custom hook.

```ts
const useValidationTextRole = ({validationState, validationText}) => {
   const [role, setRole] =useState<string | undefined>();
   const previousValidationText = useRef(validationText);
   
   useEffect(() => {
    if (previousValidationText.current !== validationText) {
      setRole(ALERT_ROLE);
    }
    previousValidationText.current = validationText;
  }, [validationText]);
  
  return role;
}
```

Then just use the hook in the components that are using ValidationText. So the role can be passed via rest props.

Reply:
This is a good idea!